source/common/stats/histogram_impl.cc

```diff
@@ -79,5 +79,64 @@ void HistogramStatisticsImpl::refresh(const histogram_t* new_histogram_ptr) {
   }
 }
 
+ParentHistogramImpl::ParentHistogramImpl(const std::string& name, Store& parent,
+                                         std::string&& tag_extracted_name, std::vector<Tag>&& tags)
+    : MetricImpl(std::move(tag_extracted_name), std::move(tags)), parent_(parent),
+      interval_histogram_(hist_alloc()), cumulative_histogram_(hist_alloc()),
+      interval_statistics_(interval_histogram_), cumulative_statistics_(cumulative_histogram_),
+      used_(false), name_(name) {}
+
+ParentHistogramImpl::~ParentHistogramImpl() {
+  hist_free(interval_histogram_);
+  hist_free(cumulative_histogram_);
+}
+
+void ParentHistogramImpl::recordValue(uint64_t value) {
+  used_ = true;
+  hist_insert_intscale(interval_histogram_, value, 0, 1);
+  parent_.deliverHistogramToSinks(*this, value);
+}
+
+void ParentHistogramImpl::merge() {
+  hist_accumulate(cumulative_histogram_, &interval_histogram_, 1);
+  interval_statistics_.refresh(interval_histogram_);
+  cumulative_statistics_.refresh(cumulative_histogram_);
+  hist_clear(interval_histogram_);
+}
+
+// TODO(mergeconflict): figure out the best way to eliminate duplication between this and
+// bucketSummary below, and the implementations in ThreadLocalParentHistogramImpl.
+const std::string ParentHistogramImpl::quantileSummary() const {
```

Comment:
nit: I don't usually qualify a by-value return object with 'const' -- I think it doesn't really add anything.

Reply:
Yeah, but it's defined that way in the `ParentHistogram` interface.